modules/kinesis/src/it/scala/com/snowplowanalytics/snowplow/enrich/kinesis/EnrichKinesisSpec.scala

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (c) 2022-2022 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.enrich.kinesis
+
+import scala.concurrent.duration._
+
+import cats.effect.{Blocker, IO}
+import cats.effect.concurrent.Ref
+
+import fs2.Stream
+
+import retry.syntax.all._
+import retry.RetryPolicies
+
+import cats.effect.testing.specs2.CatsIO
+
+import org.specs2.mutable.Specification
+import org.specs2.specification.AfterAll
+
+import com.snowplowanalytics.snowplow.analytics.scalasdk.Event
+
+import com.snowplowanalytics.snowplow.enrich.common.fs2.config.io.{Input, Output}
+
+class EnrichKinesisSpec extends Specification with AfterAll with CatsIO {
+
+  override protected val Timeout = 10.minutes
+
+  val retryPolicy =
+    RetryPolicies.limitRetriesByCumulativeDelay(5.minutes, RetryPolicies.capDelay[IO](2.second, RetryPolicies.fullJitter[IO](1.second)))
+
+  def afterAll: Unit = Containers.localstack.stop()
+
+  "enrich-kinesis" should {
+    "initialize with the minimal config" in {
+      for {
+        enrich <- IO(Containers.enrichMinimal)
+        isRunning = IO(enrich.isRunning())
```

Comment:
> When does isRunning return true?

When the container is running, so right after `start()`.

Here we don't care when the container is running or not. The only thing that we care about is to see `Running Enrich` in the logs, which means that the configuration has been successfully parsed.

What is probably misleading is that we wait for the container to be stopped (`isRunning = false`, that's what the retries do), and then when it's stopped we check that the logs contain `Running Enrich`. I agree that it would be more clear to have a `WaitStrategy` that waits for this `Running Enrich` log, and right away check again that the logs contain it in the test, because this test isn't about checking that Enrich exits. i'll update.

> And it can be also used for SIGTERM case instead of sleeping

Yes. Although for SIGTERM there is no other choice than having a retry that checks that Enrich exits after receiving the SIGTERM.